sdk/storage/azure-storage-blob-changefeed/azure/storage/blob/changefeed/_models.py

```diff
@@ -0,0 +1,407 @@
+    def __init__(self, client, page_size, start_time=None, end_time=None, cf_cursor=None):
+        self.client = client
+        self.page_size = page_size
+        self.unprocessed_segment_paths = []
+        self.current_segment = None
+        self.start_time = start_time
+        self.end_time = end_time
+        self._initialize(cf_cursor=cf_cursor)
+        self.cursor = None
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        change_feed = []
+        remaining_to_load = self.page_size
+
+        if not self.current_segment:
+            raise StopIteration
+
+        # reset the current segment page size. The page size which was set to remaining_to_load in the last call
+        # could be very small
+        self.current_segment.page_size = self.page_size
+
+        while len(change_feed) < self.page_size and self.current_segment:
+            try:
+                page_of_events = next(self.current_segment)
+                # extend the current page of events
+                change_feed.extend(page_of_events)
+                remaining_to_load -= len(page_of_events)
+                self.cursor = {"segment_path": self.current_segment.segment_path.name,
+                               "segment_cursor": self.current_segment.cursor}
+            except StopIteration:
+                self.cursor = None
+                self.current_segment = self._get_next_segment(remaining_to_load)
+
+        if not change_feed:
+            raise StopIteration
+        return change_feed
+
+    next = __next__  # Python 2 compatibility.
+
+    def _initialize(self, cf_cursor=None):
+
+        self.unprocessed_segment_paths = collections.deque(self.client.list_blobs(
+            name_starts_with=SEGMENT_COMMON_PATH))
+
+        # if start_time is specified, remove all segments earlier than start_time
+        if self.start_time:
+            while self.unprocessed_segment_paths and \
+                    self._is_earlier_than_start_time(self.unprocessed_segment_paths[0]):
+                self.unprocessed_segment_paths.popleft()
+
+        # if change_feed_cursor is specified, start from the specified segment
+        if cf_cursor:
+            while self.unprocessed_segment_paths[0].name != cf_cursor['segment_path']:
+                self.unprocessed_segment_paths.popleft()
+
+        self.current_segment = self._get_next_segment(self.page_size,
+                                                      segment_cursor=cf_cursor['segment_cursor'] if cf_cursor else None)
+
+    def _get_next_segment(self, page_size, segment_cursor=None):
+        if self.unprocessed_segment_paths:
+            segment_path = self.unprocessed_segment_paths.popleft()
+            if self.end_time and self._is_later_than_end_time(segment_path):
+                self.unprocessed_segment_paths = []
+                return None
+            return Segment(self.client, segment_path, page_size, segment_cursor)
+        return None
```

Comment:
why do we need the last return None?

Reply:
`self.current_segment = self._get_next_segment(.....)`
because if there's no Segment we need to mark self.current_segment = None